rules/falco_rules.yaml

```diff
@@ -3420,3 +3420,35 @@
 # Application rules have moved to application_rules.yaml. Please look
 # there if you want to enable them by adding to
 # falco_rules.local.yaml.
+
+
+- macro: rocketmq_check_netty_server
+  condition: (proc.name contains NettyServer and proc.env contains ROCKETMQ)
+
+- macro: rocketmq_check_payload_for_conf_update
+  condition: ((fd.typechar='4' or fd.typechar='6') and evt.type=read and evt.buffer contains 'code":25' and evt.buffer contains serializeTypeCurrentRPC and evt.buffer contains JSON)
+
+- rule: Apache Rocketmq CVE-2023-33246 Exploit Detection
+  desc: >
+    CVE-2023-33246 is exploited at the function used to update configuration & can be detected by looking at the buffer for broker code-25 and serializeTypeCurrentRPC function.
+  condition: >
+    rocketmq_check_netty_server and rocketmq_check_payload_for_conf_update and evt.dir=<
+  output: exploit detected CVE-2023-33246 (event=%evt.type server_ip=%fd.sip server_port=%fd.sport proto=%fd.l4proto parent=%proc.pname process=%proc.name command=%proc.cmdline container_id=%container.id)
```

Comment:
```
enabled: false
```

Plus can the entire rule and the respective macros be later commented out similar to how we do it for existing rules that use read? We can do this as last step to first have the CI check syntax ...